### PR TITLE
Make the .bzl files available as input to a bzl_library

### LINKED ISSUE #565: Increase visibility of all .bzl files to public.
It's unfortunate but the concept of `private` breaks other tools (eg. Stardoc).
https://github.com/bazelbuild/stardoc/issues/93

### PR DESCRIPTION
**Pull request by aiuto**
Make the .bzl files available as input to a bzl_library, without having to resort to the invasivness of a shadow tree of just the .bzl files without the other distributable files.

Fixed #565

### CODE REVIEW

**Comment by achew22** [comment]
Out of curiosity, why is a shadow tree so onerous? It is how every other language that I'm aware of in the Bazel ecosystem works. Can you help me understand why this is an improvement over the "normal" model of having an `x_library` with `deps` on other `x_library`s?

File: pkg/BUILD [hunk 1]
```diff
@@ -42,7 +43,23 @@ filegroup(
...
+
+starlark_library(
+    name = "bzl_srcs",
+    srcs = [
+        ":standard_package",
+        "//pkg/private:standard_package",
+        "//pkg/private/deb:standard_package",
+        "//pkg/private/tar:standard_package",
+        "//pkg/private/zip:standard_package",
+        "//pkg/releasing:standard_package",
```
Comment (guw) [inline pkg/BUILD:60]: This needs `"//pkg:pkg.bzl"`

File: pkg/private/make_starlark_library.bzl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+def _make_starlark_library(ctx):
+    direct = []
+    transitive = []
+    for src in ctx.attr.srcs:
+        if StarlarkLibraryInfo in src:
+            transitive.append(src[StarlarkLibraryInfo])
+        else:
+            for file in src[DefaultInfo].files.to_list():
+                if file.path.endswith(".bzl"):
+                    print(file.path)
```
Comment (duarten) [inline pkg/private/make_starlark_library.bzl:17]: Leftover debug statement?

File: pkg/BUILD (hunk 1 above)
Comment (aiuto) [reply pkg/BUILD:60]: Added.

File: pkg/private/make_starlark_library.bzl (hunk 2 above)
Comment (aiuto) [reply pkg/private/make_starlark_library.bzl:17]: Thanks for catching that.

**Comment by aiuto** [comment]
> Out of curiosity, why is a shadow tree so onerous? It is how every other language that I'm aware of in the Bazel ecosystem works. 

There are a few things I don't like about it.
- We're not actually happy with the pattern within the Bazel code itself. We're trying to get away from that a bit with the recently added `subpackages` method. 
- I have to do it twice, once for the sources (including the python) to go in the distribution, and then again for just the .bzl files.  The point of this CL is to have one human written mechanism to gather the distribution tree and then a mechanical way to unsure the .bzl files are available.
- the need for the .bzl files is because of the current implementation of stardoc, not for a conceptual reason. The design we would like to move to is one where bazel directly emits the doc information rather than our standalone parsing. That would change the flavor of the stardoc rule to require only the file you want emitted.

**Pull request merged**